src/modules/types.rs

```diff
@@ -14,14 +14,26 @@ pub enum Type {
     Generic
 }
 
+impl Type {
+    pub fn is_subset(&self, of: &Type) -> bool {
+        match (self, of) {
+            (Type::Array(l), Type::Array(r)) => {
+                **l != Type::Generic && **r == Type::Generic
+            }
+            (l, Type::Failable(r)) => !matches!(l, Type::Failable(_)) && *l == **r,
```

Comment:
How would `Failable(Generic)` work here? Can we check if a type is a subset of any Failable?

Reply:
Good catch. Right now returning a concrete array in a failable function, with a generic array as the return type, leads to:
```
ERROR: Return type does not match function return type
```

Reply:
Fixed and added a test for it

Reply:
P.S. Actually included the test this time